test/integration/accounts.js

```diff
@@ -64,12 +65,41 @@ describe('Accounts', function () {
     return wallet.balance(await wallet.address()).should.eventually.be.a('string')
   })
 
-  it('Spend 50% of balance', async () => {
-    const balance = await wallet.balance(await wallet.address())
+  describe('transferFunds', async () => {
+    const spend = async fraction => {
+      const balanceBefore = new BigNumber(await wallet.balance(await wallet.address()))
+      const { tx } = await wallet.transferFunds(fraction, receiver)
+      const balanceAfter = new BigNumber(await wallet.balance(await wallet.address()))
+      return {
+        balanceBefore,
+        balanceAfter,
+        amount: new BigNumber(tx.amount),
+        fee: new BigNumber(tx.fee)
+      }
+    }
+
+    it('spends 0% of balance', async () => {
+      const { balanceBefore, balanceAfter, amount } = await spend(0)
+      balanceBefore.should.be.not.eql(balanceAfter)
+      amount.isZero().should.be.equal(true)
+    })
+
+    it('spends 25% of balance', async () => {
+      const { balanceBefore, balanceAfter, amount, fee } = await spend(0.25)
+      amount.times(4).should.be.eql(balanceBefore)
+      balanceAfter.plus(amount).plus(fee).should.be.eql(balanceBefore)
+    })
+
```

Comment:
Can you also add a test for spending 50% of the balance? Not sure why it was removed.

Reply:
In general, I see 4 classes of tests of the percentage option:
- value out of range (no covered currently)
- minimum value
- maximum value
- value in the range from 0 to 100%

25% and 50% are both from the last class. I have chosen 25 because it looks a bit more interesting than 50%, but now I think would be better to choose a completely random one, like 68.97% 😄 Also, these tests are quite expensive so I don't see reasons to include several values from the same class of tests. How would like to proceed with that?

Reply:
> In general, I see 4 classes of tests of the percentage option:
> 
> - value out of range (no covered currently)
> 
> - minimum value
> 
> - maximum value
> 
> - value in the range from 0 to 100%
> 
> 
> 
> 25% and 50% are both from the last class. I have chosen 25 because it looks a bit more interesting than 50%, but now I think would be better to choose a completely random one, like 68.97% 😄 Also, these tests are quite expensive so I don't see reasons to include several values from the same class of tests. How would like to proceed with that?

68.97% looks nice 🤔✌️

Reply:
This is a good test, applying it I have found a rounding bug.